### PR TITLE
Add babel-eslint as parser

### LINKED ISSUE #415: [eslint-config-terra] Add babel-eslint as the parser
# Feature Request
Add babel-eslint as the parser for eslint-config-terra to be able to support things like optional chaining.

### PR DESCRIPTION
**Pull request by yuderekyu**
### Summary

Add babel-eslint as our eslint parser. This allows us to use current js functionality, such as optional chaining. Closes #415 

### Additional Details

@cerner/terra

[CONTRIBUTORS.md]: ../blob/main/CONTRIBUTORS.md
[License]: ../blob/main/LICENSE

### CODE REVIEW

File: packages/eslint-config-terra/package.json [hunk 1]
```diff
@@ -39,5 +39,8 @@
   },
   "peerDependencies": {
     "eslint": "^6.1.0"
+  },
+  "devDependencies": {
+    "babel-eslint": "^10.1.0"
```
Comment (emilyrohrbough) [inline packages/eslint-config-terra/package.json]: Should this be a hard dependency so it's provided when eslint-config-terra is installed? Does this lock us into a specific babel version?

Comment (yuderekyu) [reply packages/eslint-config-terra/package.json]: https://github.com/cerner/terra-toolkit/pull/424/files#diff-ac4371ccbae65851f05c94f13356f88eR33

arg good catch, yeah this should be a hard dependency so downstream projects pull it it. the parser expects a babel version of at least 7.2 via peer dependency https://github.com/babel/babel-eslint/blob/master/package.json

File: packages/eslint-config-terra/CHANGELOG.md [hunk 2]
```diff
@@ -1,110 +1,108 @@
...
+* Changed
+  * Updated 'jsx-a11y/label-has-for' to require id or nested input for label mapping
+## 1.0.1 - (June 19, 2018)
+
+* Changed
+  * Minor update to ReadMe
+
+## 1.0.0 - (June 18, 2018)
+
+>>>>>>> main
```
Comment (emilyrohrbough) [inline packages/eslint-config-terra/CHANGELOG.md]:
```suggestion

```

**Review by emilyrohrbough — APPROVED** [review]

File: packages/eslint-config-terra/package.json [hunk 3]
```diff
@@ -30,6 +30,7 @@
     "lint": "eslint --ext .js,.jsx ."
   },
   "dependencies": {
+    "babel-eslint": "^10.1.0",
```
Comment (StephenEsser) [inline packages/eslint-config-terra/package.json:33]: Something we should keep an eye on, this looks like it's close to deprecation in favor of the @babel/eslint-parser

https://github.com/babel/babel-eslint#note-babel-eslint-is-now-babeleslint-parser-and-has-moved-into-the-babel-monorepo

Comment (StephenEsser) [reply packages/eslint-config-terra/package.json:33]: I'm a little on the fence about this being a hard dependency or a peer dependency, eslint itself is a peer dependency. Consumers get it for free as a hard dependency, but I don't know if there's ever been compat issues between eslint and babel-eslint.

File: packages/eslint-config-terra/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,8 @@
 # Changelog
 
 ## Unreleased
+* Added
```
Comment (StephenEsser) [inline packages/eslint-config-terra/CHANGELOG.md:5]: I believe this is a lint warning. The sections need a line break between the section and first entry. There's a VS code plugin you can download the double check and keep an eye on these things

**Review by StephenEsser — APPROVED** [review]

File: packages/eslint-config-terra/package.json (hunk 3 above)
Comment (emilyrohrbough) [reply packages/eslint-config-terra/package.json:33]: Should we pull in `@babel/eslint-parser` from the get go??

Comment (StephenEsser) [reply packages/eslint-config-terra/package.json:33]: That was my initial thought too, but there's a note indicating it to still be experimental until babel 8.

https://github.com/babel/babel/tree/master/eslint/babel-eslint-parser#this-project-is-still-experimental-and-will-be-released-with-babel-v8-you-can-track-our-progress-here-please-use-babel-eslint-in-the-meantime

However, you look at the linked progress issue it has a release in babel v7. So I'm not sure what the real status of the library is:

https://github.com/babel/babel/issues/10752

Comment (yuderekyu) [reply packages/eslint-config-terra/package.json:33]: @emilyrohrbough @StephenEsser i believe we're in the clear to switch to `@babel/eslint-parser`. those docs are from the master branch, but it looks like they also switched to `main` months ago

https://github.com/babel/babel/tree/main/eslint/babel-eslint-parser

Comment (yuderekyu) [reply packages/eslint-config-terra/package.json:33]: arg, actually there is a peer dependency on eslint v7: https://github.com/babel/babel/blob/main/eslint/babel-eslint-parser/package.json#L30

unless anyone has objections, let's keep the dependency pointed to `babel-eslint` for now, then update to the scoped package on our eslint v7 uplift

https://github.com/cerner/terra-toolkit/issues/427

File: packages/eslint-config-terra/CHANGELOG.md (hunk 4 above)
Comment (yuderekyu) [reply packages/eslint-config-terra/CHANGELOG.md:5]: https://github.com/cerner/terra-toolkit/pull/424/commits/540a20f8e567cd2ce41fec3c48b752efe19253df

**Review by mjhenkes — APPROVED** [review]

**Pull request merged**